SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -283,7 +278,7 @@ void USpatialActorChannel::RetireEntityIfAuthoritative()
 				NetDriver->ActorSystem->RetireEntity(EntityId, Actor->IsNetStartupActor());
 			}
 		}
-		else if (bCreatedEntity) // We have not gained authority yet
+		else if (NetDriver->ActorSystem->IsCreateEntityRequestInFlight(EntityId)) // We have not gained authority yet
```

Comment:
Turns out I can't check the view here to see if the entity doesn't exist locally, because we are responding to a removeEntityOp here, so the view would have already been updated with the entity deleted.
I decided to implement part of @samiwh's recommendation to have a list of createEntityRequests in flight, but didn't see a good way to do it at the view level, so decided to do it at the ActorSystem level.

Open to any suggestions here - as it stands currently, it's theoretically more efficient to have a flag on the actor channel conveying this information.

Reply:
This should be at the actor system level. Although rather than it be actual system command requests, I'd make it more abstractly - what is the worker currently creating. The start point is when you decide you want to create a new entity and the end point is when you receive authority over that entity. The actual create request itself starts after that and ends before that.